File: app/src/main/java/com/ardev/idroid/base/BaseFragment.java
package com.ardev.idroid.base;
import android.view.MenuInflater;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import androidx.appcompat.view.menu.MenuBuilder;
import androidx.core.view.MenuProvider;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.ViewModelProvider;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.ardev.idroid.app.MainViewModelProvider;
import com.ardev.idroid.ui.main.MainActivity;
import com.ardev.idroid.ui.main.MainViewModel;
import com.google.android.material.appbar.MaterialToolbar;



public  class BaseFragment extends Fragment {

MaterialToolbar toolbar;
protected MainViewModel mainViewModel;
	public BaseFragment(String path) {

		Bundle args = new Bundle();
		args.putString("path", path);
		setArguments(args);
	}
	

	public String getPath() {
		return requireArguments().getString("path");
	}

	@Override
	public void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
			mainViewModel = MainViewModelProvider.getViewModel();
		
				
				
				
				
		}
		
			@Override
	public void onViewCreated(View view, Bundle savedInstanceState) {
   		 super.onViewCreated(view, savedInstanceState);
	 
		requireActivity().addMenuProvider(new MenuProvider() {
            @Override
            public void onCreateMenu(@NonNull Menu menu, @NonNull MenuInflater menuInflater) {
         if (menu instanceof MenuBuilder) {
			MenuBuilder menuBuilder = (MenuBuilder) menu;
			menuBuilder.setOptionalIconsVisible(true);
		}
			menu.clear();
		setMenu(menu);
                
               
                
            }

            @Override
            public boolean onMenuItemSelected(@NonNull MenuItem menuItem) {
            
            //onMenuSelected(menuItem);
 
                return false;
            }
        }, getViewLifecycleOwner(), Lifecycle.State.RESUMED);
	}
	
	 
	
	 public MaterialToolbar getToolbar() {
        if (toolbar == null) toolbar = ((MainActivity) requireActivity()).getToolbar();
        return toolbar;
    }
	
	
	protected void setMenu(Menu menu) {}
	//protected void onMenuSelected(MenuItem menuItem) {}
	}

File: app/src/main/java/com/ardev/idroid/base/BaseFragment.kt
package com.ardev.idroid.base

import android.view.MenuInflater
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.widget.LinearLayoutCompat
import androidx.viewbinding.ViewBinding
import androidx.appcompat.view.menu.MenuBuilder
import androidx.core.view.MenuProvider
import androidx.fragment.app.Fragment
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import android.os.Bundle


	abstract class BaseFragment<VB : ViewBinding, VM : ViewModel> : Fragment() {

	private var _binding: VB? = null
    protected val binding get() = _binding!!
    protected abstract val viewModel: VM


    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
       
requireActivity().addMenuProvider( object: MenuProvider {
			override fun onCreateMenu(menu: Menu, menuInflater: MenuInflater) {
	menu.clear()
	setMenu(menu)
 }
			override fun onMenuItemSelected(menuItem: MenuItem): Boolean = false

        }, viewLifecycleOwner, Lifecycle.State.RESUMED)
        
     observeViewModel() 
    

    }


    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
    	_binding =  getViewBinding(inflater, container)
        return binding.root  
    }
	override fun onDestroy() {
	super.onDestroy()
	_binding = null
	}

	protected fun setMenu(menu: Menu) {}
	protected open fun observeViewModel() {}
    protected abstract fun getViewBinding(inflater: LayoutInflater, container: ViewGroup?): VB




}